Adding migration guide for on-prem to konnect (#7482)
Kong/docs.konghq.com: app/konnect/getting-started/migration.md

---

> title: Migrating from Kong Gateway On-Premises to Kong Konnect

Reviewer: From the title I'm expecting a step-by-step guide on how to migrate. The steps written feel more abstract, and I understand that everyones case is different but if we want to use illustrative steps maybe we can reorder them so that they express a recommended order of operations? 

Like: 

1. Export config
2. Modify Config 
3. Import config
4. Setup RBAC 
5. Migrate Data planes 
Then you have extra information that may not apply to everyone like: 
Handling plugins

---

> The success of {{site.base_gateway}} can be attributed to its flexible deployment model, 
> exceptional performance, broad extensibility, and Open Source and Enterprise licensing models.
> This has resulted in a large deployment base over a variety of environments and topologies. 
> 
> As organizations grow and scale they need for more advanced capabilities while lowering 
> operational complexity. Features such as strong [multi-tenancy](https://konghq.com/blog/enterprise/multi-tenancy), 
> federated API management, advanced security integrations, and others. With {{site.konnect_product_name}}, 
> users have access to a full featured API management platform which provides managed components and 
> builds on the lessons learned from deep customer usage of {{site.base_gateway}} and API management best practices. 

Reviewer: ```suggestion
Kong Konnect offers advanced features such as strong [multi-tenancy](https://konghq.com/blog/enterprise/multi-tenancy), 
federated API management, advanced security integrations. Migrating to Konnect from Kong Gateway is a straight forward process, designed to ensure a smooth migration experience.
```

The introduction distracts me from the purpose of the document which I believe to be: 
1. Konnect offers features that would make a user want to migrate from On-prem to Konnect
2. Migrating to Konnect is straight forward.

This document will provide a guide for {{site.base_gateway}} users looking to migrate to {{site.konnect_product_name}}. 

> ## Migration Guide
> 
> The following will provide key details for completing a successful migration from {{site.base_gateway}} on-premises to 
> {{site.konnect_product_name}}. This document focuses on migrating a [Hybrid](/gateway/latest/production/deployment-topologies/hybrid-mode/) 
> or [Traditional](/gateway/latest/production/deployment-topologies/traditional/)
> deployment to Konnect, for other deployment modes see the following:
> 
> * [{{site.kic_product_name}} (KIC)](/kubernetes-ingress-controller/latest/)
>     
>     If you run KIC on premises, migrating to Konnect is straightforward. 
>     The [Kong Ingress Controller for Kubernetes Assocation](/konnect/gateway-manager/kic/)
>     documentation provides details on linking your KIC deployment to a KIC based Control Plane
>     in {{site.konnect_product_name}}.
> 
> * [DB-less mode](/gateway/latest/production/deployment-topologies/db-less-and-declarative-config/)
> 
>     If you're running a DB-less deployment and interested in migrating to Konnect, Kong recommends migrating to 
>     Hybrid mode deployment to take advantage of the full capabilities of Kong Konnect. Please reach out to
>     a Kong representative for assistance with this migration.
> 

Reviewer: ```suggestion
## Migration Guide

The following will provide key details for completing a successful migration from {{site.base_gateway}} on-premises to 
{{site.konnect_product_name}}. 

<Insert Flow-chart of path or steps to migration

This document focuses on migrating a [Hybrid](/gateway/latest/production/deployment-topologies/hybrid-mode/) 
or [Traditional](/gateway/latest/production/deployment-topologies/traditional/)
deployment to Konnect, for other deployment modes see the following:

* [{{site.kic_product_name}} (KIC)](/kubernetes-ingress-controller/latest/)
    
    If you run KIC on premises, migrating to Konnect is straightforward. 
    The [Kong Ingress Controller for Kubernetes Assocation](/konnect/gateway-manager/kic/)
    documentation provides details on linking your KIC deployment to a KIC based Control Plane
    in {{site.konnect_product_name}}.

* [DB-less mode](/gateway/latest/production/deployment-topologies/db-less-and-declarative-config/)

    If you're running a DB-less deployment and interested in migrating to Konnect, Kong recommends migrating to 
    Hybrid mode deployment to take advantage of the full capabilities of Kong Konnect. Please reach out to
    a Kong representative for assistance with this migration.

```

Reviewer 2: Agree with diagram. Resolving this comment though, as I'll just add a diagram directly instead of suggesting a placeholder.

> ## Migration Guide
> 
> The following will provide key details for completing a successful migration from {{site.base_gateway}} on-premises to 
> {{site.konnect_product_name}}. This document focuses on migrating a [Hybrid](/gateway/latest/production/deployment-topologies/hybrid-mode/) 
> or [Traditional](/gateway/latest/production/deployment-topologies/traditional/)
> deployment to Konnect, for other deployment modes see the following:
> 
> * [{{site.kic_product_name}} (KIC)](/kubernetes-ingress-controller/latest/)
>     
>     If you run KIC on premises, migrating to Konnect is straightforward. 
>     The [Kong Ingress Controller for Kubernetes Assocation](/konnect/gateway-manager/kic/)
>     documentation provides details on linking your KIC deployment to a KIC based Control Plane
>     in {{site.konnect_product_name}}.
> 
> * [DB-less mode](/gateway/latest/production/deployment-topologies/db-less-and-declarative-config/)
> 
>     If you're running a DB-less deployment and interested in migrating to Konnect, Kong recommends migrating to 
>     Hybrid mode deployment to take advantage of the full capabilities of Kong Konnect. Please reach out to
>     a Kong representative for assistance with this migration.
> 

Reviewer 2: ```suggestion
## Migration guide

This guide provides key details for completing a successful migration from a self-managed {{site.base_gateway}} to
{{site.konnect_short_name}}. It focuses on migrating a [hybrid](/gateway/latest/production/deployment-topologies/hybrid-mode/) 
or [traditional](/gateway/latest/production/deployment-topologies/traditional/) to {{site.konnect_short_name}}.

<!--vale off-->
{% mermaid %}
flowchart TD
A{Which deployment 
topology are you 
starting with?}
B(Traditional)
C(Hybrid)
D(KIC)
E(DB-less)

A-->B & C & D & E

B & C --> F["Migrate to Konnect
(use this guide)"]
E --> H["Migrate to hybrid mode
(reach out to support for help)"] --> F
D --> G[<a href="/konnect/gateway-manager/kic/">Link your KIC deployment to a
KIC-based control plane</a>]
{% endmermaid %}
<!--vale on-->

> **Figure 1**: For traditional and hybrid deployments, you can migrate directly to {{site.konnect_short_name}} by following this guide. For DB-less deployments, migrate to hybrid first, then follow this guide. For KIC deployments, migrate to a KIC-based control plane.
```

> ### Role Based Access Controls

Reviewer 2: ```suggestion
### Role Based Access Controls (RBAC)
```

Both {{site.base_gateway}} and {{site.konnect_product_name}} provide frameworks for controlling security and
data access. 

{{site.base_gateway}} provides a traditional Role Based Access Control (RBAC) system
to manage administrators and users of the API gateway system. {{site.konnect_product_name}} provides a robust 

> multi-layer Identity and Access Management (IAM) system including organizations, teams and roles. 

Reviewer 2: ```suggestion
multi-layer Identity and Access Management (IAM) system including organizations, teams, and roles. 
```

> Konnect also provides integrations with IdP providers via OIDC allowing you to map centrally managed teams to 
> Konnect based roles.

Reviewer 2: ```suggestion
{{site.konnect_product_name}} also provides integrations with IdP providers via OIDC, allowing you to map centrally managed teams to 
{{site.konnect_product_name}}-based roles.
```

{{site.base_gateway}}'s RBAC system does not map directly to the IAM system provided by {{site.konnect_product_name}}. 

> Users migrating from {{site.base_gateway}} on-premises to {{site.konnect_product_name}} have typically 
> chosen to use Konnect's IdP integrations and take advantage of their existing IdP solution and Konnect team 
> based mappings instead migrating their {{site.base_gateway}} on-premises RBAC configuration directly.

Reviewer 2: ```suggestion
When migrating from a self-managed {{site.base_gateway}} to {{site.konnect_short_name}}, 
we recommend using {{site.konnect_short_name}}'s IdP integrations. 
You can take advantage of your existing IdP solution and {{site.konnect_short_name}}'s 
team-based mappings instead of migrating your self-managed {{site.base_gateway}} 
RBAC configuration directly.
```

The following are the general steps for setting up IAM in {{site.konnect_product_name}} for your migration:

> * [Sign up](https://konghq.com/products/kong-konnect/register?utm_medium=referral&utm_source=docs) for 
>   Konnect (if necessary), and use the [Org Switcher](/konnect/org-management/org-switcher/) 
>   to create or select the Organization you are going to migrate your on-premises deployment to. 

Reviewer 2: ```suggestion
1. [Sign up](https://konghq.com/products/kong-konnect/register?utm_medium=referral&utm_source=docs) for 
  {{site.konnect_short_name}} (if necessary), and use the [Org Switcher](/konnect/org-management/org-switcher/) 
  to create or select the organization you are going to migrate your self-managed deployment to. 
```

> * [Set up single sign-on (SSO) access to Konnect using an existing IdP provider](/konnect/org-management/oidc-idp/)

Reviewer 2: ```suggestion
2. [Set up single sign-on (SSO) access to {{site.konnect_short_name}} using an existing IdP provider](/konnect/org-management/oidc-idp/)
```

> * [Create teams](/konnect/org-management/teams-and-roles/manage/) in Konnect or use 

Reviewer 2: ```suggestion
3. [Create teams](/konnect/org-management/teams-and-roles/manage/) in {{site.konnect_short_name}} or use 
```

  [predefined teams](/konnect/org-management/teams-and-roles/teams-reference/) to create your desired organizational structure.

> * For any custom teams, [assign the appropriate roles](/konnect/org-management/teams-and-roles/manage/#edit-roles-for-a-team) 
>   from the predefined list of available roles in Konnect.

Reviewer 2: ```suggestion
4. For any custom teams, [assign the appropriate roles](/konnect/org-management/teams-and-roles/manage/#edit-roles-for-a-team) 
  from the predefined list of available roles in {{site.konnect_short_name}}.
```

> * Use the Konnect IdP Team Mappings feature to 
>   [map the Konnect teams to your IdP provider groups](/konnect/org-management/okta-idp/#map-konnect-teams-to-okta-groups).

Reviewer 2: ```suggestion
5. Use the {{site.konnect_short_name}} IdP Team Mappings feature to 
  [map the{ {site.konnect_short_name}} teams to your IdP provider groups](/konnect/org-management/okta-idp/#map-konnect-teams-to-okta-groups).
```

> ### Workspaces -> Control Planes

Reviewer: ```suggestion
## Migrating from Workspaces to control planes
```

I would make this an H2, and make the subtopics H3's. The information ehre is solid enough to stand alone.

Reviewer 2: Let's hold off on changing header levels for now. Let's leave it nested, along with RBAC, inside the migration guide for this first iteration. I do agree with the retitling though.
```suggestion
### Migrating from workspaces to control planes
```

> {{site.base_gateway}} supports configuration isolation using
> [Workspaces](/gateway/latest/kong-enterprise/workspaces/).
> {{site.konnect_product_name}}'s model is more advanced and is solved 
> using [Control Planes](/konnect/gateway-manager/#control-planes) which are managed, virtual, and lightweight
> components used to isolate configuration. 
> 
> When migrating to {{site.konnect_product_name}}, you will create a Control Plane design that 
> best fits your goals, which may or may not mirror the number of Workspaces you
> use on-premises.

Reviewer 2: ```suggestion
{{site.base_gateway}} supports configuration isolation using
[workspaces](/gateway/latest/kong-enterprise/workspaces/).
{{site.konnect_product_name}}'s model is more advanced and uses
[control planes](/konnect/gateway-manager/#control-planes) which are managed, virtual, and lightweight
components used to isolate configuration. 

When migrating to {{site.konnect_product_name}}, you will create a control plane design that 
best fits your goals, which may or may not mirror the number of workspaces you
use in your self-managed deployment.
```

> #### Control Plane Design
> 
> If you currently use a single Workspace in your on-premises installation,
> you can simply create a matching Control Plane with the same name. Alternatively, 
> you may choose to take the opportunity to re-organize your single Workspace configuration 
> into multiple Control Planes if there is a clear separation of concerns in your gateway configuration.
> 
> If you're using multiple workspaces in your on-premises installation, the most straightforward
> approach is to create a Control Plane for each workspace, but you may choose to reorganize your 
> design during the migration.

Reviewer 2: ```suggestion
#### Control plane design

If you currently use a single workspace in your self-managed installation,
you can simply create a matching control plane with the same name. Alternatively, 
you can choose to take the opportunity to re-organize your single workspace configuration 
into multiple control planes if there is a clear separation of concerns in your gateway configuration.

If you're using multiple workspaces in your self-managed installation, the most straightforward
approach is to create a control plane for each workspace, but you may choose to reorganize your 
design during the migration.
```

#### Multi-tenancy

> {{site.base_gateway}} Workspaces also provide a way to share runtime infrastructure across isolated configurations.
> With {{site.konnect_product_name}}, this is achieved using 
> [Control Plane Groups](/konnect/gateway-manager/control-plane-groups/). Control Planes can be added
> and removed from Control Plane Groups and setup to mirror your existing mutli-tenant Workspace configuration. 
> Once you have Control Plane Groups setup, you can connect data plane instances to the group creating
> a share data plane infrastructure among the constituent Control Planes.

Reviewer 2: ```suggestion
{{site.base_gateway}} workspaces provide a way to share runtime infrastructure across isolated configurations.
With {{site.konnect_product_name}}, this is achieved using 
[control plane groups](/konnect/gateway-manager/control-plane-groups/). Control planes can be added to
and removed from control plane groups, and you can set them up to mirror your existing multi-tenant workspace configuration. 

With control plane groups set up, you can connect data plane instances to each group, creating
a shared data plane infrastructure among the constituent control planes.
```

> #### Control Plane Management 
> 
> Managing Control Planes and Control Plane Groups in {{site.konnect_product_name}} can be 
> achieved by using the [Konnect UI](/konnect/gateway-manager/), the
> [Konnect Control Planes API](/konnect/api/control-planes/latest/), or the 
> [{{site.konnect_product_name}} Terraform Provider](https://registry.terraform.io/providers/Kong/konnect/latest).

Reviewer 2: ```suggestion
#### Control plane management 

You can manage control planes and control plane groups in {{site.konnect_product_name}} by using the [Konnect UI](/konnect/gateway-manager/), the
[Konnect Control Planes API](/konnect/api/control-planes/latest/), or the 
[{{site.konnect_product_name}} Terraform Provider](https://registry.terraform.io/providers/Kong/konnect/latest).
```

> #### Example Workspace Migration
> 
> The following provides an example set of steps for migrating a small multi-Workspace setup to 
> {{site.konnect_product_name}}.
> 
> {:.note}
> > **Note**: Instructions on this page are not step-by-step guides. They are meant to be illustrative of the general steps 
> you can perform to migrate Workspaces. The examples use [decK](/deck/latest/) and some well known command line tools for 
> querying APIs and manipulating text output. See the [jq](https://jqlang.github.io/jq/) and [yq](https://github.com/mikefarah/yq) 
> tool pages for more information.

Reviewer 2: ```suggestion
#### Example workspace migration

The following provides an example set of steps for migrating a small multi-workspace setup to 
{{site.konnect_product_name}}.

{:.note}
> **Note**: Instructions on this page are not step-by-step guides. They are meant to illustrate the general steps 
you can perform to migrate workspaces. The examples use [decK](/deck/latest/) and some well known command-line tools for 
querying APIs and manipulating text output. See the [jq](https://jqlang.github.io/jq/) and [yq](https://github.com/mikefarah/yq) 
tool pages for more information.
```

> We can start by querying the [Admin API](/gateway/latest/admin-api/) of our on-premises installation to get 
> a list of workspaces for a particular {{site.base_gateway}} installation:

Reviewer 2: ```suggestion
Query the [Admin API](/gateway/latest/admin-api/) of your self-managed installation to get 
a list of workspaces for a particular {{site.base_gateway}} deployment:
```

```sh
curl -s localhost:8001/workspaces | jq -r '.data.[].name'
```

> And we'll receive a response with a name for each Workspace:

Reviewer 2: ```suggestion
You will receive a response with a name for each workspace:
```

```text
default
inventory
sales
```

> Our example on-premises installation has three total workspaces: The `default` workspace, `inventory`, and `sales`. 

Reviewer 2: ```suggestion
In this example, the self-managed installation has three total workspaces: the `default` workspace, `inventory`, and `sales`. 
```

> {:.note}
> > **Note**: {{site.base_gateway}} provides a `default` Workspace, and similarly {{site.konnect_product_name}} provides a 
> `default` Control Plane. Neither of these can be deleted, so migrating the `default` Workspace to the `default` Control Plane 
> is a logical choice.

Reviewer 2: ```suggestion
{:.note}
> **Note**: {{site.base_gateway}} provides a `default` workspace, and similarly {{site.konnect_product_name}} provides a 
`default` control plane. Neither of these can be deleted, so migrating the `default` workspace to the `default` control plane 
is a logical choice.
```

> Create a new Control Plane for each non-default Workspace:

Reviewer 2: ```suggestion
Create a new control plane for each non-default workspace:
```

> ```txt

Reviewer 2: ```suggestion
```sh
```

curl -s -H "Authorization: Bearer ${KONNECT_PAT}" \
  --request POST \
  --url https://us.api.konghq.com/v2/control-planes \
  --header 'Content-Type: application/json' \
  --header 'accept: application/json' \
  --data '{"name":"inventory","description":"CP for the Inventory team.","cluster_type":"CLUSTER_TYPE_CONTROL_PLANE"}'
```

> ```txt

Reviewer 2: ```suggestion
```sh
```

curl -s -H "Authorization: Bearer ${KONNECT_PAT}" \
  --request POST \
  --url https://us.api.konghq.com/v2/control-planes \
  --header 'Content-Type: application/json' \
  --header 'accept: application/json' \
  --data '{"name":"sales","description":"CP for the Sales team.","cluster_type":"CLUSTER_TYPE_CONTROL_PLANE"}'
```

> Login into the Konnect UI and validate the new Control Planes.

Reviewer 2: ```suggestion
Log in into the {{site.konnect_short_name}} UI and validate the new control planes.
```

### Plugins

> Konnect supports the majority of plugins available to {{site.base_gateway}}, however,
> {{site.base_gateway}} is ran in Hybrid Mode which limits support for plugins that require direct access
> to a database. 
> 
> {:.note}
> > **Note**: Konnect also provides Dedicated Cloud Gateways which 
> further limit plugins that require specialized software agents running on the dataplane hosts. 
> 
> In order to migrate plugins from on-premises to Konnect you should review 
> [Konnect Compatibility page](/konnect/compatibility/) for your usage of unsupported
> plugins. Additionally, review if certain configuration values are unsupported which will require additional
> changes to your configuration.

Reviewer 2: ```suggestion
{{site.konnect_short_name}} supports the majority of plugins available to {{site.base_gateway}}. Since {{site.konnect_short_name}} runs in hybrid mode, this limits support for plugins that require direct access
to a database. 

{:.note}
> **Note**: {{site.konnect_short_name}} also provides Dedicated Cloud Gateways, which 
further limit plugins that require specialized software agents running on the data plane hosts. 

To migrate plugins from a self-managed deployment to {{site.konnect_short_name}}, review 
[Konnect Compatibility page](/konnect/compatibility/) to check for supported and unsupported plugins.
Also review any plugin configuration values, as certain values are unsupported in {{site.konnect_short_name}} and may require additional
changes to your configuration.
```

> ### Custom Plugins
> 
> {{site.konnect_product_name}} supports custom plugins with similar restrictions to pre-built plugins. Given 
> that {{site.base_gateway}} runs in a Hybrid deployment mode, custom plugins may not access a database directly
> and can not provide a custom Admin API endpoint. See the Konnect documentation for more details
> on [Custom Plugin support](/konnect/gateway-manager/plugins/#custom-plugins) requirements.
> 
> Migrating custom plugins to Konnect requires uploading and associating your custom plugin's `schema.lua` file to 
> the desired Control Plane. This can be done using the 
> [Konnect UI](/konnect/gateway-manager/plugins/add-custom-plugin/) or the 
> [Konnect Control Planes Config API](/konnect/api/control-plane-configuration/latest/#/Custom%20Plugin%20Schemas/list-plugin-schemas).
> 
> Just like in on-premises deployments, the custom plugin code must be distributed to the data plane instances.
> 
> {:.note}
> > **Note**: Konnect's Dedicated Cloud Gateways can support custom plugins
> but currently require a manual deployment process involving {{site.base_gateway}}'s support personal. Contact your Kong representative
> for more information.

Reviewer 2: ```suggestion
### Custom plugins

{{site.konnect_short_name}} supports custom plugins with similar restrictions to pre-built plugins. Since {{site.konnect_short_name}} runs in a hybrid deployment mode, custom plugins can't access a database directly
and can't provide custom Admin API endpoints. See the {{site.konnect_short_name}} documentation for more details
on [custom plugin support](/konnect/gateway-manager/plugins/#custom-plugins) requirements.

Migrating custom plugins to {{site.konnect_short_name}} requires uploading and associating your custom plugin's `schema.lua` file to 
the desired control plane. This can be done using the 
[Konnect UI](/konnect/gateway-manager/plugins/add-custom-plugin/) or the 
[Konnect Control Planes Config API](/konnect/api/control-plane-configuration/latest/#/Custom%20Plugin%20Schemas/list-plugin-schemas).

Just like in self-managed deployments, the custom plugin code must be distributed to the data plane instances.

{:.note}
> **Note**: {{site.konnect_short_name}}'s Dedicated Cloud Gateways can support custom plugins
but currently require a manual deployment process involving {{site.base_gateway}}'s support team. 
Contact your Kong representative for more information.
```

> ### {{site.base_gateway}} Configuration

Reviewer 2: ```suggestion
### {{site.base_gateway}} configuration
```

> Typically the {{site.base_gateway}} configuration is migrated to {{site.konnect_product_name}} 
> using [decK](/deck/latest/guides/konnect/), the declarative management tool for {{site.base_gateway}}
> configurations.

Reviewer 2: ```suggestion
We recommend migrating {{site.base_gateway}} configuration to {{site.konnect_product_name}} 
using [decK](/deck/latest/guides/konnect/), the declarative management tool for {{site.base_gateway}}
configurations.
```

> The general process for migrating the configuration involves "dumping" your existing on-premises configuration
> to a local file, modifying the configuration slightly to remove any workspace specific metadata,
> and then synchronizing the configuration to your desired Control Plane in {{site.konnect_product_name}}.

Reviewer 2: ```suggestion
The general process for migrating the configuration involves "dumping" your existing  self-managed configuration
to a local file, modifying the configuration slightly to remove any workspace-specific metadata,
and then synchronizing the configuration to your desired control plane in {{site.konnect_short_name}}.
```

> Continuing the example from earlier, use `deck` to dump the configuration of each Workspace to a local file:

Reviewer 2: ```suggestion
Continuing the example from earlier, use `deck` to dump the configuration of each workspace to a local file:
```

```sh
deck gateway dump --workspace default --output-file default.yaml
deck gateway dump --workspace inventory --output-file inventory.yaml
deck gateway dump --workspace sales --output-file sales.yaml
```

> When using `deck` to dump the configuration, the output file will include the Workspace name in the configuration.
> To remove the `_workspace` key from the configuration, you can use the `yq` tool to remove the key:

Reviewer 2: ```suggestion
When using `deck` to dump the configuration, the output file will include the workspace name in the configuration.
To remove the `_workspace` key from the configuration, you can use the `yq` tool:
```

```sh
yq -i 'del(._workspace)' default.yaml
yq -i 'del(._workspace)' inventory.yaml
yq -i 'del(._workspace)' sales.yaml 

> ```

Author: IMHO this isn't necessary.  `yq` and `jq` are standard in developer workflows that manipulate files of these formats.  Given this is a "hint" to how to perform the migration and not a recipe for any automated workflow, I don't think the use of the tool is concerning.

> Synchronize the configuration to the Control Planes using decK configured with the 
> proper Control Plane name and the Konnect access token:

Reviewer 2: ```suggestion
Synchronize the configuration to the control planes using decK configured with the 
proper cpontrol plane name and the {{site.konnect_short_name}} access token:
```

```sh
deck gateway sync --konnect-token "${KONNECT_PAT}" --konnect-control-plane-name default default.yaml
deck gateway sync --konnect-token "${KONNECT_PAT}" --konnect-control-plane-name inventory inventory.yaml
deck gateway sync --konnect-token "${KONNECT_PAT}" --konnect-control-plane-name sales sales.yaml
```

> In addition to the above process of using decK for the migration, {{site.konneect_product_name}} provides
> other tools that could be used for migrating your configuration. Each tool will require a different process,
> reference their documentation for more information.

Reviewer 2: ```suggestion
In addition to decK, {{site.konnect_short_name}} provides
other tools that could be used for migrating your configuration. Each tool requires a different process. See their documentation for more information:
```

* [Konnect Control Planes Config API](/konnect/api/control-plane-configuration/latest/)
* [{{site.konnect_product_name}} Terraform Provider](/konnect/reference/terraform/)

> ### Data planes
> 
> The recommended approach for migrating your data plane instances to {{site.konnect_product_name}} is to
> create new data plane instances connected to your Control Plane, validate their configuration and connectivity,
> and then decommission the on-premises data plane instances.
> 
> The {{site.konnect_product_name}} documentation provides details on 
> [{{site.base_gateway}} installation options](/konnect/gateway-manager/data-plane-nodes/). The easiest
> way to deploy new data planes is using the Konnect Gateway Manager, which provides integrated 
> launchers for popular operating systems and compute platforms. 

Reviewer 2: ```suggestion
### Data planes

The recommended approach for migrating your data plane instances to {{site.konnect_product_name}} is to
create new data plane instances connected to your control plane, validate their configuration and connectivity,
and then decommission the self-managed data plane instances.

The {{site.konnect_product_name}} documentation provides details on 
[{{site.base_gateway}} installation options](/konnect/gateway-manager/data-plane-nodes/). The easiest
way to deploy new data planes is using the {{site.konnect_short_name}} Gateway Manager, which provides integrated 
launchers for popular operating systems and compute platforms. 
```

### APIOps

> Konnect users will find that there are additional options for managing the API deployment lifecycle on
> compared to {{site.base_gateway}} on-premises. 

Reviewer 2: ```suggestion
In {{site.konnect_short_name}}, there are additional options for managing the API deployment lifecycle, as compared to {{site.base_gateway}}.
```

> If you're using [deck](/deck/latest) to manage your {{site.base_gateway}} configuration, you can continue to use

Reviewer 2: ```suggestion
* **decK**: If you're using [deck](/deck/latest) to manage your {{site.base_gateway}} configuration, you can continue to use
```

the tool to managing your {{site.konnect_product_name}} configuration. The decK CLI 

> [supports Konnect Control Plane configuration](/deck/latest/guides/konnect/)

Reviewer 2: ```suggestion
[supports {{site.konnect_short_name}} control plane configuration](/deck/latest/guides/konnect/)
```

> by providing additional flags that configure the tool to connect to a particular Control Plane using access tokens.

Reviewer 2: ```suggestion
by providing additional flags that configure the tool to connect to a particular control plane using access tokens.
```

> Additionally, {{site.konnect_product_name}} provides a Terraform provider for managing a full Konnect deployment 
> including Control Planes, Control Plane Groups, data plane configuration and more. The 
> [{{site.konnect_product_name}} Terraform Provider](/konnect/reference/terraform/) can be used independently
> or in conjunction with decK to manage your API deployment lifecycle.
> 
> The [{{site.kgo_product_name}}](/gateway-operator/latest/) is also available for teams 
> that desire to use standardized Kubernetes APIs to manage their Konnect deployments.

Reviewer 2: ```suggestion
* **Terraform**: Kong provides a Terraform provider for managing a full {{site.konnect_short_name}} deployment, 
including control planes, control plane groups, data plane configuration, and more. The 
[{{site.konnect_short_name}} Terraform Provider](/konnect/reference/terraform/) can be used independently
or in conjunction with decK to manage your API deployment lifecycle.

* **{{site.kgo_product_name}}**:  The [{{site.kgo_product_name}} (KGO)](/gateway-operator/latest/) is available for teams 
that want to use standardized Kubernetes APIs to manage their {{site.konnect_short_name}} deployments.
```

## Next steps

> If you are interested in assistance with migrating from {{site.base_gateway}} on-premises to 
> {{site.konnect_product_name}}, please contact a Kong field representative.

Reviewer 2: ```suggestion
If you are interested in assistance with migrating from a self-managed {{site.base_gateway}} to 
{{site.konnect_short_name}}, contact a Kong field representative.
```
